network/builder/src/builder.rs

```diff
@@ -90,8 +90,9 @@ pub struct NetworkBuilder {
     // TODO(philiphayes): better support multiple listening addrs
     listen_address: NetworkAddress,
     advertised_address: Option<NetworkAddress>,
-    seed_peers: HashMap<PeerId, Vec<NetworkAddress>>,
-    trusted_peers: Arc<RwLock<HashMap<PeerId, x25519::PublicKey>>>,
+    trusted_peers: Arc<RwLock<HashMap<PeerId, HashSet<x25519::PublicKey>>>>,
```

Comment:
I'd say it's the correct data structure to use in any case (it'd be annoying to implement a set on top of a vec). Also you're supposed to only talk to a fixed number of validators (in the hundreds) so it's not that much.

Reply:
definitely agree a Vec would be more efficient backing since each set is very small. in fact, we could probably get away with [tinyvec](https://docs.rs/tinyvec/0.3.3/tinyvec/) to remove an extra indirection. we're just missing some convenient wrappers to emulate set-like api, which i would probably tackle later, when we have more time to tackle optimizations.

Reply:
Yeah, I was just posing the question.  A set is the right way to go definitely.